src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -2894,12 +2894,14 @@ List<JCStatement> blockStatement() {
                         int lookahead = 2;
                         int balance = 1;
                         boolean hasComma = false;
+                        boolean hasTypeArgs = false;
```

Comment:
For consideration, may also balance out the `<`/`>`, to cover cases like this:
```
    private static int t(int i) {
        yield((Map<String, String>) null, 2);
    }

    private static void yield(Map<String, String> m, int j) {
        
    }
```

Not critical, as this is error recovery anyway, so for consideration only.

Reply:
Addressed. Thanks @lahodaj